kong/plugins/api-response-merger/body_transformer.lua

```diff
@@ -283,6 +286,12 @@ function _M.transform_json_body(keys_to_extend, upstream_body, http_config)
     local config = resource.config
     -- if response is an array we should use set_in_table_arr
     -- multiple resources
+
+    if resource_body == nil and config.resource_id_optional == true then
+      kong.log.warn('Resource id is not optional ', resource_key, ' api: ', config.api.url, ' res_body: nil')
```

Comment:
I don't understand this error message - don't we have resource_id_optional == true in the line above?